### PR TITLE
Fix check for missing service registration to throw InvalidOperationException instead of NullReferenceException.

### LINKED ISSUE #28: When service is not registered, unhelpful NullReferenceException is thrown instead of InvalidOperationException
## Describe the Bug

When using `AutofacWebServiceHostFactory` to create a service, when that service is not registered in the container, results in an unhelpful NRE, instead of throwing the intended InvalidOperationException (https://github.com/autofac/Autofac.Wcf/blob/1987b3ef48136430673ee72a4fa2320e618384d8/src/Autofac.Integration.Wcf/DefaultServiceImplementationDataProvider.cs#L85).

## Steps to Reproduce

See this minimal working example project: https://github.com/JarkoDubbeldam/Autofac.Integraction.Wcf.BugRepro/blob/master/BugRepro/ServiceHostFactory.cs

## Expected Behavior

I'd expect a more helpful error :) 

## Exception with Stack Trace

```text
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Autofac.Integration.Wcf
  StackTrace:
   at Autofac.Integration.Wcf.DefaultServiceImplementationDataProvider.GetServiceImplementationData(String value)
   at Autofac.Integration.Wcf.AutofacHostFactory.CreateServiceHost(String constructorString, Uri[] baseAddresses)
   at BugRepro.ServiceHostFactory.CreateServiceHost(String constructorString, Uri[] baseAddresses) in C:\Users\jdubbeldam\source\repos\BugRepro\BugRepro\ServiceHostFactory.cs:line 17

```

## Dependency Versions

```text
<packages>
  <package id="Autofac" version="6.0.0" targetFramework="net472" />
  <package id="Autofac.Wcf" version="6.0.0" targetFramework="net472" />
  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.1.0" targetFramework="net472" />
  <package id="System.Buffers" version="4.5.1" targetFramework="net472" />
  <package id="System.Diagnostics.DiagnosticSource" version="4.7.1" targetFramework="net472" />
  <package id="System.Memory" version="4.5.4" targetFramework="net472" />
  <package id="System.Numerics.Vectors" version="4.5.0" targetFramework="net472" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.5.3" targetFramework="net472" />
  <package id="System.Threading.Tasks.Extensions" version="4.5.2" targetFramework="net472" />
</packages>
```

## Additional Info

It appears to me that [this](https://github.com/autofac/Autofac.Wcf/blob/1987b3ef48136430673ee72a4fa2320e618384d8/src/Autofac.Integration.Wcf/DefaultServiceImplementationDataProvider.cs#L83) statement can never be true, due to `serviceRegistration` not being nullable.

### PR DESCRIPTION
**Pull request by JarkoDubbeldam**
Fixes #28.

### CODE REVIEW

File: src/Autofac.Integration.Wcf/DefaultServiceImplementationDataProvider.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public virtual ServiceImplementationData GetServiceImplementationData(string val
                 }
             }
 
-            if (serviceRegistration == null)
+            if (serviceRegistration == null || serviceRegistration == default)
```
Comment (JarkoDubbeldam) [inline src/Autofac.Integration.Wcf/DefaultServiceImplementationDataProvider.cs:83]: I feel like the null check is unnecessary, due to serviceRegistration being a non-nullable struct. I left it in there to be sure.

Comment (tillig) [reply src/Autofac.Integration.Wcf/DefaultServiceImplementationDataProvider.cs:83]: I think it's fine. I think you're correct, though, that it'll never be hit.

**Review by tillig — APPROVED** [review]
LGTM. Thanks!

**Comment by tillig** [comment]
I have #30 in right now to update the build to the latest tooling and increment the semver (still targeting .NET 4.7.2, just _building_ with newer tooling). I'll cut a release that includes this as part of the completion of that PR.

**Pull request merged**